src/main/java/org/fisco/bcos/channel/client/Service.java

```diff
@@ -915,6 +936,167 @@ public void run() {
+
+        SDKVersion version = new SDKVersion();
+        message.setData(objectMapper.writeValueAsBytes(version));
+
+        logger.info(
+                "connection established，send sdk channel supported highest protocol to the connection , seq:{}, sdk channel protocol version:{}",
+                message.getSeq(),
+                version);
+
+        ByteBuf out = ctx.alloc().buffer();
+        message.writeHeader(out);
+        message.writeExtra(out);
+
+        ctx.writeAndFlush(out);
+    }
+
+    public void sendHeartbeatMessage(ChannelHandlerContext ctx) {
+
+        Message message = new BcosMessage();
+        message.setSeq(UUID.randomUUID().toString().replaceAll("-", ""));
+        message.setResult(0);
+        message.setType((short) 0x13);
+
+        HeartBeatParser heartBeatParser = new HeartBeatParser(getUsageChlVersion(ctx));
+
+        try {
+            message.setData(heartBeatParser.encode("0"));
+        } catch (JsonProcessingException e) {
+            logger.error(" write json failed, message is {}", e.getMessage());
+            return;
+        }
+
+        ByteBuf out = ctx.alloc().buffer();
+        message.writeHeader(out);
+        message.writeExtra(out);
+
+        ctx.writeAndFlush(out);
+    }
+
+    public void onReceiveHeartbeat(ChannelHandlerContext ctx, Message msg) {
+
+        String content = "";
+
+        HeartBeatParser heartBeatParser = new HeartBeatParser(getUsageChlVersion(ctx));
+        String data = new String(msg.getData());
+        try {
+            BcosHeartbeat bcosHeartbeat = heartBeatParser.decode(data);
+            // logger.trace(" heartbeat packet, heartbeat is {} ", bcosHeartbeat);
+            int heartBeat = bcosHeartbeat.getHeartBeat();
+            content = String.valueOf(heartBeat);
+        } catch (UnsupportedEncodingException e) {
+            logger.error("heartbeat packet cannot be parsed, data is {}", data);
+        } catch (Exception e) {
+            logger.error("heartbeat packet exception, data is {}", data);
+        }
+
+        if ("0".equals(content)) {
+            logger.trace("heartbeat packet，send heartbeat packet back");
+            Message response = new Message();
+
+            response.setSeq(msg.getSeq());
+            response.setResult(0);
+            response.setType((short) 0x13);
+
+            try {
+                response.setData(heartBeatParser.encode("1"));
+            } catch (JsonProcessingException e) {
+                logger.error(" write json failed, message is {} ", e.getMessage());
+                return;
+            }
+
+            ByteBuf out = ctx.alloc().buffer();
+            response.writeHeader(out);
+            response.writeExtra(out);
+
+            ctx.writeAndFlush(out);
+        } else if ("1".equals(content)) {
+            logger.trace("heartbeat response");
+        }
+    }
+
+    public org.fisco.bcos.channel.protocol.Version getUsageChlVersion(ChannelHandlerContext ctx) {
```

Comment:
the function name is ugly, plz do not use an abbreviation, getChannelVersion is more available

Reply:
> the function name is ugly, plz do not use an abbreviation, getChannelVersion is more available
done